support/cas-server-support-aup-couchdb/src/main/java/org/apereo/cas/config/CasAcceptableUsagePolicyCouchDbConfiguration.java

```diff
@@ -0,0 +1,62 @@
+package org.apereo.cas.config;
+
+import org.apereo.cas.aup.AcceptableUsagePolicyRepository;
+import org.apereo.cas.aup.CouchDbAcceptableUsagePolicyRepository;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.couchdb.core.CouchDbConnectorFactory;
+import org.apereo.cas.couchdb.core.ProfileCouchDbRepository;
+import org.apereo.cas.ticket.registry.TicketRegistrySupport;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.ektorp.impl.ObjectMapperFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * This is {@link CasAcceptableUsagePolicyCouchDbConfiguration} that stores AUP decisions in a CouchDb database.
+ *
+ * @author Timur Duehr
+ * @since 6.0.0
+ */
+@Configuration("casAcceptableUsagePolicyCoucbDbConfiguration")
+@EnableConfigurationProperties(CasConfigurationProperties.class)
+@Slf4j
+public class CasAcceptableUsagePolicyCouchDbConfiguration {
+    @Autowired
+    @Qualifier("defaultTicketRegistrySupport")
+    private TicketRegistrySupport ticketRegistrySupport;
+
+    @Autowired
+    private CasConfigurationProperties casProperties;
+
+    @Autowired
+    private ObjectMapperFactory objectMapperFactory;
+
+    @Bean
```

Comment:
Please include `ConditionalOnMissingBean` annotations for all beans.

Reply:
Also consider using ObjectProvider for injected fields.

Reply:
Will do.